flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPoolTest.java

```diff
@@ -518,4 +528,176 @@ public void go() throws Exception {
 			globalPool.destroy();
 		}
 	}
+
+	/**
+	 * Tests {@link NetworkBufferPool#isAvailable()}, verifying that the buffer availability is
+	 * correctly maintained and the future callback is correctly processed.
+	 */
+	@Test
+	public void testBufferAvailabilityAndFutureCallback() throws Exception {
+		final int numBuffers = 10;
+		final int numberOfSegmentsToRequest = 5;
+		final Duration requestSegmentsTimeout = Duration.ofSeconds(10L);
+
+		final NetworkBufferPool globalPool = new NetworkBufferPool(
```

Comment:
We could use the simple constructor `NetworkBufferPool(int numberOfSegmentsToAllocate, int segmentSize, int numberOfSegmentsToRequest)` for tests instead.